### ISSUE TITLE
Image regeneration should be optional.

### ISSUE THREAD

**Issue by phj123** [woocommerce/woocommerce#19478]
The in 3.3 introduced WC image regeneration engine should not fire without user permission.

This problem, it will override by image optimization plugins, already optimized images.
This will bring users a lot of images that have to be reoptimized again.
It will cause a lot of extra resources regarding image optimization and also extra unwanted images on your server.

Also regenerating images just out of the blue with a lot of users not even knowing it happened can cause image SEO issues.

What if someone that is unknowing just tries out how some image sizes will appear?

Bottom line an important task as regeneration of images should be a user his own choice and optional!

**Comment by GlennMartin1** [comment]
In my case, I use TinyPNG for image compression.

Since WC 3.3, I set my Customizer ratio to 3:2 (same as it was prior to WC 3.3). Then WC regenerated thumbnails. Fine.

Since WC regenerated thumbnails, I now have almost 10,000 images that need to be recompressed (according to TinyPNG).

Is that normal/expected?

**Comment by claudiulodro** [comment]
[Here's a one-line code-snippet for deactivating the feature if you don't want or need it.](https://gist.github.com/lukecav/904cc76e5ce89e9d62fcf6ec22f44495) I don't think it makes sense to deactivate it by default since it would only cause issues on sites that are doing custom image handling already, and this is a very small amount of total WooCommerce sites. 

> Since WC regenerated thumbnails, I now have almost 10,000 images that need to be recompressed (according to TinyPNG). Is that normal/expected?

It depends on how many products you have in your store and how your TinyPNG integration works. There are a few image sizes that get regenerated for each product and each size of regenerated image will probably need to be recompressed.

Closing this issue because image regeneration is optional (see snippet above). It's just on by default.

**Comment by phj123** [comment]
Ok forget about optimization for a moment and review this opinion please:

My concern is most users don't even know it will regenerate! The WP customer tab is broadly used for trying out your theme and how it will look.

If someone tries out different image sizes it will regenerate every time and they are left with a lot of extra images on their server. The Responsive WP image function will still pick up all those images an output them in there srcset image HTML.

If someone has tried out a lot of sizes close to another he will be left with a lot of unnecessary unwanted images sizes on his server that moreover also will be appearing unnecessary in there in srcset image HTML output.

I'm very curious, what are your thoughts on this?
